src/Chainweb/Pact/PactService/ExecBlock.hs

```diff
@@ -144,7 +146,7 @@ execBlock currHeader plData pdbenv = do
       [] -> return ()
       errs -> throwM $ TransactionValidationException $ errs
 
-    !results <- go miner trans
+    !results <- go miner trans >>= throwOnGasFailure
```

Comment:
This simply replicates previous behavior that a buy gas failure occuring in validate-block renders the block invalid. In the past this would simply `throwM` directly from the failure in `TransactionExec.applyCmd` (`applyBuyGas`).